.github/workflows/branch-deploy.yml

```diff
@@ -19,61 +18,42 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           token: ${{ github.token }}
+
       - name: get branch name
         id: extract_branch
         run: |
           tag='pr${{ github.event.pull_request.number }}'
           echo "tag=${tag}" >> $GITHUB_OUTPUT
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
           java-version: '17'
           distribution: 'zulu'
           cache: 'maven'
+
       - name: Build
```

Comment:
an action like this is present in 4 workflows:
- `build-public-image`
- `branch-deploy`
- `release`
- `separate_env_public_create`
The point of #117 was to reduce the amount of these things, can we extract them?